framework/src/auxkernels/GapValueAux.C

```diff
@@ -104,16 +101,14 @@ GapValueAux::computeValue()
       gap_value = _moose_var.getValue(pinfo->_elem, side_phi);
   }
   else
-  {
-    if (_warnings)
-    {
-      std::stringstream msg;
-      msg << "No gap value information found for node ";
-      msg << current_node->id();
-      msg << " on processor ";
-      msg << processor_id();
-      mooseWarning(msg.str());
-    }
-  }
+    mooseDoOnce(mooseWarning("No gap value information found for node ",
+                             current_node->id(),
+                             " on processor ",
+                             processor_id(),
+                             " at coordinate ",
+                             Point(*current_node),
+                             ". The computed result will most likely be wrong.\n A likely cause is "
```

Comment:
There are lots of cases where you might not project to another surface. This message would be misleading.